src/rastervision/tagging/tasks/validation_eval.py

```diff
@@ -65,9 +65,10 @@ def plot_prediction(generator, all_x, y_true, y_pred,
         imgs.append(ir_im)
         titles.append('IR')
 
-        ndvi_im = all_x[:, :, dataset.ndvi_ind]
-        imgs.append(ndvi_im)
-        titles.append('NDVI')
+        if dataset.ndvi_ind:
```

Comment:
This is problematic since ndvi_ind could be zero, in which case the condition would be false. Instead, use `dataset.ndvi_ind is not None`